lib/ruby_parser.yy

```diff
@@ -1783,23 +1782,11 @@ opt_block_args_tail: tCOMMA block_args_tail
 
      symbol_list: none
                     {
-                      result = s(:array)
+                      result = new_symbol_list
                     }
                 | symbol_list word tSPACE
                     {
-                      list, sym, _ = val
-
-                      case sym[0]
-                      when :dstr then
-                        sym[0] = :dsym
-                      when :str then
-                        sym = s(:lit, sym.last.to_sym)
-                      else
-                        debug20 24
-                        sym = s(:dsym, "", result)
```

Comment:
I moved all this logic to `new_symbol_list_entry`. But, while moving the code I noticed this line. I'm not super clear on what `result` is on this line. I have zero familiarity with reading this file (other than pattern matching), so I wasn't clear if `result` was using the value on line 1786 above, or if this was just a bug and this line was never actually getting hit.